exercises/word-search/example.js

```diff
@@ -0,0 +1,146 @@
+class WordSearch {
+  constructor(grid) {
+    this.grid = grid;
+  }
+
+  find(words) {
+    return words
+      .map(word => ({[word]: findWordInAnyDirection(word, this.grid)}))
+      .reduce((acc, oneWord) => Object.assign(acc, oneWord), {});
+  }
+}
+
+function findWordInAnyDirection(word, grid) {
+  return searchHorizontally({word, grid})
+    || flipCoordinates(searchHorizontally({word, grid: flipGrid(grid)}))
+    || searchDiagonally({word, grid})
+    || searchDiagonally({word, grid, fromTop: false});
+}
+
+function searchHorizontally({word, grid}) {
+  let rowIndex = 0;
+  let startCol;
+  let start;
+  let end;
+
+  const getCoords = () => [[rowIndex + 1, startCol], [rowIndex + 1, startCol + word.length - 1]];
+
+  const getStartCol = (word) => 1 + grid[rowIndex].indexOf(word);
+
+  while (rowIndex < grid.length) {
+    startCol = getStartCol(word);
+    if (startCol) {
+      [start, end] = getCoords();
+    } else {
+      startCol = getStartCol([...word].reverse().join(''));
+      if (startCol) {
+        [end, start] = getCoords();
+      }
+    }
+    if (start && end) {
+      return {start, end};
+    }
+    rowIndex++;
+  }
+}
```

Comment:
Should this function return something explicitly? Even `return undefined`?

Not sure, just to make it explicit, and probably the `or` operator at `findWordInAnyDirection` would be a bit clearer .

Reply:
I could certainly add that if you think it's better to have it explicit.  I just suggested to a contributor that he take his out!!  :)

Reply:
On one hand, I would like to see all returning points. On the other hand, returning `undefined` is not very helpful and may confuse the reader.